### DISCUSSION TITLE
Add relevant Repairs Issue Registry entries to config entry and device diagnostics

### DISCUSSION THREAD

**Discussion by bachya** [home-assistant/architecture#807] [General] [answered]
## Background

The Repairs Issue Registry allows users to learn about (and potentially triage) issues with their Home Assistant instance (ahead of time and as they occur). In two particular cases:

1. Issues whose `is_fixable` attribute is set to `False`
2. Issues that implement [the default `ConfirmRepairFlow`](https://github.com/home-assistant/core/blob/d39ed0cde4d5e9d34849fb1df2d0ea1f32d4e1e2/homeassistant/components/repairs/issue_handler.py#L25-L50)

...users are asked to confirm their understanding and acceptance of the issue (by ignoring it in case 1 and "dismissing" it in case 2).

With this possibility arises a potential difficulty: the user raises a bug report related to that confirmed issue, forgetting that Repairs previously notified them. This could lead to an arduous diagnosis process that wastes an excessive amount of time for the responding dev. The dev could ultimately short-circuit the diagnosis with a simple way to say, "You confirmed that you understood XYZ would happen two versions ago...").

## Proposal

To solve this issue, I propose that we modify the default diagnostics payload (for both config entries and devices) to contain any Repairs Issue Registry entries that share the same domain. For example, if the user had accepted notification of a forthcoming RainMachine entity replacement, their config entry/device diagnostics would look like this (extra data removed):

```json
{
  "home_assistant": {},
  "custom_components": {},
  "integration_manifest": {
    "domain": "rainmachine",
    "name": "RainMachine",
    "config_flow": true,
    "documentation": "https://www.home-assistant.io/integrations/rainmachine",
    "requirements": [
      "regenmaschine==2022.09.1"
    ],
    "codeowners": [
      "@bachya"
    ],
    "iot_class": "local_polling",
    "homekit": {
      "models": [
        "Touch HD",
        "SPK5"
      ]
    },
    "zeroconf": [
      {
        "type": "_http._tcp.local.",
        "name": "rainmachine*"
      }
    ],
    "loggers": [
      "regenmaschine"
    ],
    "is_built_in": true
  },
  "integration_issues": [
    {
      "created": "2022-09-17T23:04:33.202016+00:00",
      "dismissed_version": "2022.10.0.dev0",
      "domain": "rainmachine",
      "is_persistent": true,
      "issue_id": "replaced_old_entity_binary_sensor.home_freeze_protection",
      "breaks_in_ha_version": "2022.12.0",
      "data": null,
      "is_fixable": true,
      "issue_domain": null,
      "learn_more_url": null,
      "severity": "warning",
      "translation_key": "deprecated_service_single_alternate_target",
      "translation_placeholders": {
        "old_entity_id": "binary_sensor.home_freeze_protection",
        "replacement_entity_id": "switch.home_freeze_protection_enabled",
      }
    },
  ],
  "data": {}
}
```

Suppose a bug report arose in which a user didn't understand that such an entity replacement had occurred. In that case, these diagnostics could quickly demonstrate that the user had seen and dismissed the Repairs Issue notification (allowing the responding dev to close the issue expediently).

## Considerations

For completeness, I recommend that the diagnostics include _all_ Repairs Issue Registry entries that match the diagnostics domain. However, the diagnostics payload could become extensive over time, depending on how many such entries exist. I don't see that as a critical problem to be addressed now, but including it in case I'm mistaken.

**Comment by thecode** [comment]
I think this is a good idea and will save time when checking issues

**Comment by allenporter** [comment]
Agree this makes sense and could be useful.

On the subject of automatically including things in diagnostics from other components, there was a related discussion before: Camera currently has a `diagnostics.py` that for right now is invoked manually by integrations that want to opt in to calling it. However, you can imagine that it could be automatically for a config entry with a camera entity, or for a device with camera entities. The specifics are different than the proposal here, but the general idea of automatically including other relevant platforms or integrations for a domain, config entry, or device, seems useful.

**Reply by bachya** [reply]
Agreed: as a separate proposal, some sort of helper that includes other components/info in a diagnostics payload would be very useful.

**Comment by emontnemery** [comment, marked as answer]
The proposal was approved by the core team July 3rd

**Reply by gjohansson-ST** [reply]
Implementartion being done in https://github.com/home-assistant/core/pull/148498
